### PR TITLE
fix: add error handling fetching glossary, lint

### PR DESCRIPTION
**Pull request by anshbansal**

### CODE REVIEW

File: glossary-sync.py [hunk 1]
```diff
@@ -4,38 +4,27 @@
...
-from ruamel.yaml import YAML
-
-from datahub.ingestion.graph.client import (
-    DatahubClientConfig,
-    DataHubGraph,
-    get_default_graph,
-)
+from datahub.ingestion.graph.client import (DatahubClientConfig, DataHubGraph,
+                                            get_default_graph)
 from datahub.ingestion.source.metadata.business_glossary import (
```
Comment (treff7es) [inline glossary-sync.py:12]: Import sorting looks weird. Did you use isort and black to format the code?

File: glossary-sync.py [hunk 2]
```diff
@@ -188,26 +177,32 @@ def _glossary_term_from_datahub(
...
-    # Get all the urns in the glossary.
+    logger.info("Get all the urns in the glossary.")
     urns = list(graph.get_urns_by_filter(entity_types=["glossaryTerm", "glossaryNode"]))
+    logger.info(f"Got {len(urns)} urns")
 
-    # Hydrate them into entities.
-    entities = {urn: graph.get_entity_semityped(urn) for urn in urns}
+    logger.info("Hydrate them into entities.")
+    entities = {}
+    for urn in progressbar.progressbar(urns):
```
Comment (treff7es) [inline glossary-sync.py:186]: Why don't we collect glossaryNodes and glossaryTerm in a different dict? 
Something like:
```suggestion
    glossary_node_enitites = {}
    glossary_term_entities = {}
    for urn in progressbar.progressbar(urns):
       try:
             entity = graph.get_entity_semityped(urn)
             if guess_entity_type(urn) != "glossaryNode":
                 glossary_node_enitites[urn] = entity
             elif guess_entity_type(urn) != "glossaryTerm": 	   				    
                 glossary_term_enitites[urn] = entity
...		          		          
```
Later in the code, you don't have to check for each entity if it is glossaryTerm or glossaryNode

File: glossary-sync.py [hunk 3]
```diff
@@ -188,26 +177,32 @@ def _glossary_term_from_datahub(
...
+        try:
+            entities[urn] = graph.get_entity_semityped(urn)
+        except requests.exceptions.HTTPError as e:
+            logger.error(f"Error for urn {urn}: {e}")
```
Comment (treff7es) [inline glossary-sync.py:190]: Why can this fail? Should we retry in this case or it is fine if we just log a failure?

**Review by treff7es — APPROVED** [review]

File: glossary-sync.py (hunk 1 above)
Comment (anshbansal) [reply glossary-sync.py:12]: yes

File: glossary-sync.py (hunk 3 above)
Comment (anshbansal) [reply glossary-sync.py:190]: Fine to log error. This is corner cases for URN encoding which will need to resolved

**Pull request merged**